src/Microsoft.AspNet.OData.Shared/Formatter/ODataOutputFormatterHelper.cs

```diff
@@ -184,7 +184,8 @@ internal static bool TryGetCharSet(MediaTypeHeaderValue mediaType, IEnumerable<s
                 writeContext.SkipExpensiveAvailabilityChecks = serializer.ODataPayloadKind == ODataPayloadKind.ResourceSet;
                 writeContext.Path = path;
                 writeContext.MetadataLevel = metadataLevel;
-                writeContext.SelectExpandClause = internalRequest.Context.SelectExpandClause;
+                writeContext.SelectExpandClause = internalRequest.Context.ProcessedSelectExpandClause;
+                writeContext.QueryOptions = internalRequest.Context.QueryOptions;
```

Comment:
note: if we move the logic for setting the ProcessedSelectExpandClause when QueryOptions is set to the HttpRequestMessageProperties object, we can remove this line (as it will get automatically set when the QueryOption is set).